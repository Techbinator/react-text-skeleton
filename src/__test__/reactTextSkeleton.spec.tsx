import * as React from 'react';
import TextSkeleton from '..';
import * as TestRenderer from 'react-test-renderer';

describe('text skeleton', () => {
  it('should render the dummy text with blurred class by default', () => {
    const component = TestRenderer.create(<TextSkeleton dummyText="dummy text" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the dummy text with gradient class', () => {
    const component = TestRenderer.create(<TextSkeleton dummyText="dummy text" skeletonType="gradient" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should add class names next to the skeleton type class', () => {
    const component = TestRenderer.create(
      <TextSkeleton dummyText="dummy text" className="passed-class-name extra-class-name" skeletonType="gradient" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should display the final text if children prop is passed', () => {
    const component = TestRenderer.create(<TextSkeleton dummyText="dummy text">Final text</TextSkeleton>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
