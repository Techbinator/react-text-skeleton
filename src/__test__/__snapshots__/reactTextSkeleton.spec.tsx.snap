// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text skeleton should add class names next to the skeleton type class 1`] = `
<span
  className="gradient passed-class-name extra-class-name"
>
  dummy text
</span>
`;

exports[`text skeleton should display the final text if children prop is passed 1`] = `
<span
  className=""
>
  Final text
</span>
`;

exports[`text skeleton should render the dummy text with blurred class by default 1`] = `
<span
  className="blurred"
>
  dummy text
</span>
`;

exports[`text skeleton should render the dummy text with gradient class 1`] = `
<span
  className="gradient"
>
  dummy text
</span>
`;
